internal/status/tracker.go

```diff
@@ -0,0 +1,77 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+)
+
+type StatusTracker struct {
+	oras.Target
+	out          io.Writer
+	printLock    sync.Mutex
+	printAfter   bool
```

Comment:
Just out of curiosity, what does `printAfter` mean?

Reply:
Print before(false) or after(true) the push.

Reply:
Note: This `StatusTracker` is a workaround. Implementation will be enhanced once https://github.com/oras-project/oras-go/pull/150 is merged.